### PR TITLE
[Feature] support alter table comment

### LINKED ISSUE #20825: [Feature request] Editable comment/description for a table or database
Discussed in https://github.com/StarRocks/starrocks/discussions/19269

<div type='discussions-op-text'>

<sup>Originally posted by **akzarma** March  9, 2023</sup>
I found that we can set the comment for a table while creating it but we cannot update it anymore once it is created. (tried with ALTER TABLE command but it didn't work)

So, here are mainly 2 feature requests:
1. Make the comment of a table editable
2. Add the functionality to add a comment on a database too.</div>

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #20825

## Problem Summary(Required) ：

Support alter table comment through: 
```sql
ALTER TABLE t1 COMMENT = 'New table comment';

```

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by imay** [comment]
I think the syntax should be `alter table xxtable modify comment ""` we need a verb before the `comment`

**Comment by Astralidea** [comment]
@imay 
https://dev.mysql.com/doc/refman/8.0/en/alter-table.html
This is the standard syntax of mysql, mysql does not support
```
alter table xxtable modify comment ""
```
But I think in addition to this grammar, other grammars should also be supported, for example:
```
alter table test_table comment 'New table comment';
```
The user omitted the wording of =

**Comment by imay** [comment]
> 

Got it, thank you.

File: fe/fe-core/src/test/java/com/starrocks/sql/analyzer/AnalyzeAlterTableStatementTest.java [hunk 1]
```diff
@@ -138,4 +138,9 @@ public void testRollup() {
         analyzeSuccess("alter table t0 drop rollup test1");
         analyzeSuccess("alter table t0 drop rollup test1, test2");
     }
+
+    @Test
+    public void testAlterTableComment() {
```
Comment (imay) [inline fe/fe-core/src/test/java/com/starrocks/sql/analyzer/AnalyzeAlterTableStatementTest.java:143]: why not adding the test cases in SQL-tests

**Review by satanson — DISMISSED** [review]

**Review by silverbullet233 — DISMISSED** [review]

**Review by satanson — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Comment by qy58644781** [comment]
> @imay https://dev.mysql.com/doc/refman/8.0/en/alter-table.html This is the standard syntax of mysql, mysql does not support
> 
> ```
> alter table xxtable modify comment ""
> ```
> 
> But I think in addition to this grammar, other grammars should also be supported, for example:
> 
> ```
> alter table test_table comment 'New table comment';
> ```
> 
> The user or BI may omitted the wording of =

thank you

**Comment by jaogoy** [comment]
@mofeiatwork You need to modify the documentation.

**Pull request merged**